### PR TITLE
MOD : query.Options.FilterRights

### PR DESCRIPTION
**Pull request by nfaugout**
Renamed to .CheckRights, enlarge definition to match either GET request or POST request

### CODE REVIEW

File: Domain/RDD.Domain/Models/RestCollection.cs [hunk 1]
```diff
@@ -37,7 +37,12 @@ public virtual Task<TEntity> CreateAsync(PostedData datas, Query<TEntity> query
 
         public virtual async Task<TEntity> CreateAsync(TEntity entity, Query<TEntity> query = null)
         {
-            await CheckRightsForCreateAsync(entity);
+            query = query ?? new Query<TEntity>();
+
+            if (query.Options.CheckRights)
```
Comment (rducom) [inline Domain/RDD.Domain/Models/RestCollection.cs]: `if (query == null || query.Options.CheckRights)`
avoid one useless allocation

Comment (nfaugout) [reply Domain/RDD.Domain/Models/RestCollection.cs]: +1

**Comment by nfaugout** [comment]
@Poltuu je te laisse merger si tu es OK avec la review

**Comment by Poltuu** [comment]
boom

**Pull request merged**